boudicca.base/publisher-event-html/src/test/kotlin/base/boudicca/publisher/event/html/PublisherHtmlApplicationE2ETests.kt

```diff
@@ -0,0 +1,307 @@
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.time.OffsetDateTime
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(SpringExtension::class)
+@Tag("playwright")
+class PublisherHtmlApplicationE2ETests: E2ETestFixture() {
+
+  @LocalServerPort
+  private val port = 0
+
+  @MockBean
+  lateinit var searchServiceCaller: SearchServiceCaller
+
+  @ParameterizedTest
+  @ArgumentsSource(E2EGeneralTestData::class)
+  fun pageShouldHaveCorrectTitle(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+    assertThat(page).hasTitle("Boudicca")
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(E2ESingleEventTestData::class)
+  fun eventShouldBeVisibleInViewport(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+
+    assertThat(page.locator(".event")).isInViewport()
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(E2ESingleEventTestData::class)
+  fun eventDetailsShouldBeCorrect(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+
+    assertThat(page.locator(".event-details").first()).containsText("01.01.2023 um 13:00 Uhr")
+    assertThat(page.locator(".event-details").last()).containsText("Theater, Linz")
+    assertThat(page.locator(".event-title")).equals("Musical Event in Innenstadt")
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(SingleEventWithA11YInformation::class)
+  fun shouldDisplayA11YIconWhenTheEventIsAccessible(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+
+    assertThat(page.locator(".accessibility-details")).isVisible()
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(E2EGeneralTestData::class)
+  fun shouldListCorrectNumberOfEvents(
+    events: List<Event>,
+    filters: Map<String, List<String>>
+  ) {
+    setupSearchServiceCaller(events, filters)
+
+    page.navigate("http://localhost:$port/")
+
+    val events = page.querySelectorAll(".event")
+    val eventSize = events.size
+
+    assertTrue(eventSize == 30) { "Expected 30 events, but found $eventSize events." }
```

Comment:
for those asserts assertEquals would be more fitting

Reply:
all replaced.